### ISSUE TITLE
Avoid invoking the republisher when nothing has changed.

### ISSUE THREAD

**Issue by Stebalien** [ipfs/go-mfs#35]
We "publish" the current root on flush, even if it hasn't changed. This means we're writing it every time we stop ipfs.

**Comment by Stebalien** [comment]
Sounds reasonable but I haven't thought through all the possible code paths.

**Comment by aschmahmann** [comment]
This repository is no longer maintained and has been copied over to [Boxo](https://github.com/ipfs/boxo).  In an effort to avoid noise and crippling in the Boxo repo from the weight of issues of the past, we are closing most issues and PRs in this repo.  Please feel free to open a new issue in Boxo (and reference this issue) if resolving this issue is still critical for unblocking or improving your usecase.

You can learn more in the [FAQs for the Boxo repo copying/consolidation effort](https://github.com/ipfs/boxo/wiki/Copied-or-Migrated-Repos-FAQ).
